src/tests/cmocka/data_provider/test_dp_request.c

```diff
@@ -433,6 +495,12 @@ int main(int argc, const char *argv[])
         cmocka_unit_test_setup_teardown(test_nonexist_dom,
                                         test_setup,
                                         test_teardown),
+        cmocka_unit_test_setup_teardown(test_permit_handler,
+                                        test_setup,
+                                        test_teardown),
+        cmocka_unit_test_setup_teardown(test_deny_handler,
+                                        test_setup,
+                                        test_teardown),
     };
```

Comment:
You are not testing dp_req anymore here, can you move these tests into separate file such as test_dp_special.c or something like that?